functions/go/export-terraform/terraformgenerator/resources.go

```diff
@@ -190,24 +195,24 @@ func (resource *terraformResource) getReferencedResource(kind string, path ...st
 }
 
 func (resource *terraformResource) getParentRef(path ...string) (string, string, error) {
-	paths := [][]string{
-		{"Folder", "spec", "folderRef", "name"},
-		{"Folder", "spec", "folderRef", "external"},
-		{"Organization", "spec", "organizationRef", "external"},
-		{"Project", "metadata", "annotations", "cnrm.cloud.google.com/my-project"},
-		{"detect", "spec", "resourceRef", "external"},
-		{"detect", "spec", "resourceRef", "name"},
+	paths := []referencePath{
+		{kind: "Folder", path: []string{"spec", "folderRef", "name"}},
+		{kind: "Folder", path: []string{"spec", "folderRef", "external"}},
+		{kind: "Organization", path: []string{"spec", "organizationRef", "external"}},
+		{kind: "Project", path: []string{"metadata", "annotations", "cnrm.cloud.google.com/project-id", "name"}},
+		{path: []string{"spec", "resourceRef", "external"}},
+		{path: []string{"spec", "resourceRef", "name"}},
 	}
 
 	for _, path := range paths {
-		name := resource.GetStringFromObject(path[1:]...)
+		name := resource.GetStringFromObject(path.path...)
 		if name == "" {
 			continue
 		}
 
-		kind := path[0]
-		if kind == "detect" {
-			kind = resource.GetStringFromObject(append(path[1:len(path)-1], "kind")...)
+		kind := path.kind
+		if len(kind) <= 1 {
+			kind = resource.GetStringFromObject(append(path.path[0:len(path.path)-1], "kind")...)
```

Comment:
`path.path[0:len(path.path)-1]` is actually `path.path`.

Reply:
They're actually not equivalent, go slices are exclusive on the top end. This is getting everything in the slice except the last element.

I added a test to verify this.

Reply:
You are right. I misread it.